### PR TITLE
Load sessions in parallel

### PR DESCRIPTION
**Pull request by hmusum**
Please review only

### CODE REVIEW

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java [hunk 1]
```diff
@@ -164,8 +169,17 @@ public SessionRepository(TenantName tenantName,
...
+        ExecutorService executor = Executors.newFixedThreadPool(Math.max(8, Runtime.getRuntime().availableProcessors()),
+                                                                new DaemonThreadFactory("load-sessions-"));
+        loadLocalSessions(executor);
+        loadRemoteSessions(executor);
+        try {
+            executor.shutdown();
+            if ( ! executor.awaitTermination(1, TimeUnit.MINUTES))
+                log.log(Level.INFO, "Executor did not terminate");
+        } catch (InterruptedException e) {
+            e.printStackTrace();
```
Comment (hakonhall) [inline configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java:181]: log instead?

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java [hunk 2]
```diff
@@ -184,18 +198,23 @@ public LocalSession getLocalSession(long sessionId) {
...
+        futures.forEach(f -> {
             try {
-                createSessionFromId(Long.parseLong(session.getName()));
-            } catch (IllegalArgumentException e) {
-                log.log(Level.WARNING, "Could not load session '" +
-                        session.getAbsolutePath() + "':" + e.getMessage() + ", skipping it.");
+                long sessionId = f.get();
+                log.log(Level.INFO, () -> "Local session " + sessionId + " loaded");
+            } catch (ExecutionException | InterruptedException e) {
+                log.log(Level.WARNING, "Could not load session");
```
Comment (hakonhall) [inline configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java:215]: add sessionId to warning?

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java [hunk 3]
```diff
@@ -342,17 +361,28 @@ private boolean sessionHasExpired(Instant created, Duration expiryTime, Clock cl
...
+        List<Future<Long>> futures = new ArrayList<>();
+        for (long sessionId : getRemoteSessionsFromZooKeeper()) {
+            futures.add(executor.submit(() -> sessionAdded(sessionId)));
+        }
+        futures.forEach(f -> {
+            try {
+                long sessionId = f.get();
+                log.log(Level.INFO, () -> "Remote session " + sessionId + " loaded");
+            } catch (ExecutionException | InterruptedException e) {
+                log.log(Level.WARNING, "Could not load session");
```
Comment (hakonhall) [inline configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java:374]: Same here

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java (hunk 1 above)
Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java]: Fixed

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java (hunk 2 above)
Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java]: Fixed

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java (hunk 3 above)
Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java]: Fixed

**Review by hakonhall — APPROVED** [review]

**Pull request merged**